ast/compile.go

```diff
@@ -120,34 +120,35 @@ type Compiler struct {
-	evalMode                CompilerEvalMode
+	localvargen                              *localVarGenerator
+	moduleLoader                             ModuleLoader
+	ruleIndices                              *util.HashMap
+	stages                                   []stage
+	maxErrs                                  int
+	sorted                                   []string // list of sorted module names
+	pathExists                               func([]string) (bool, error)
+	after                                    map[string][]CompilerStageDefinition
+	metrics                                  metrics.Metrics
+	capabilities                             *Capabilities                 // user-supplied capabilities
+	imports                                  map[string][]*Import          // saved imports from stripping
+	builtins                                 map[string]*Builtin           // universe of built-in functions
+	customBuiltins                           map[string]*Builtin           // user-supplied custom built-in functions (deprecated: use capabilities)
+	unsafeBuiltinsMap                        map[string]struct{}           // user-supplied set of unsafe built-ins functions to block (deprecated: use capabilities)
+	deprecatedBuiltinsMap                    map[string]struct{}           // set of deprecated, but not removed, built-in functions
+	enablePrintStatements                    bool                          // indicates if print statements should be elided (default)
+	comprehensionIndices                     map[*Term]*ComprehensionIndex // comprehension key index
+	initialized                              bool                          // indicates if init() has been called
+	debug                                    debug.Debug                   // emits debug information produced during compilation
+	schemaSet                                *SchemaSet                    // user-supplied schemas for input and data documents
+	inputType                                types.Type                    // global input type retrieved from schema set
+	annotationSet                            *AnnotationSet                // hierarchical set of annotations
+	strict                                   bool                          // enforce strict compilation checks
+	keepModules                              bool                          // whether to keep the unprocessed, parse modules (below)
+	parsedModules                            map[string]*Module            // parsed, but otherwise unprocessed modules, kept track of when keepModules is true
+	useTypeCheckAnnotations                  bool                          // whether to provide annotated information (schemas) to the type checker
+	allowUndefinedFuncCalls                  bool                          // don't error on calls to unknown functions.
+	evalMode                                 CompilerEvalMode
+	rewriteTestRulesToCaptureUnboundDynamics bool
```

Comment:
"Dynamics" -- it's not a term that's common in ast or topdown, is it? 🤔

Reply:
True 😄, I think that trickled up from the `rewriteDynamics*()` naming convention we have in the compiler. Naming suggestions are welcome :). It's a bit too long for my liking too ..